packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -81,16 +83,29 @@ function DocSidebarItemCategory({
     if (justBecameActive && collapsed) {
       setCollapsed(false);
     }
+    if (item.link) {
+      if (Object.hasOwnProperty.call(item.link, 'type')) {
+        switch (item.link.type) {
+          case 'doc':
+            setInitialLink(item.link.id);
```

Comment:
this is not correct, the id is not the URL path, as docs can use slugs they have ids that can be totally different from their final URL.

This requires resolving the id to the doc URL path

(+ shouldn't need setState)